reporting/impl/src/main/resources/reports/templates/techReport-punchCard.ftl

```diff
@@ -306,15 +319,87 @@
             });
         }
 
+        function sortTableBySector(sector) {
+            // cleanup the previous sort classes
+            var groupedHeaders = $('.headersGroup td').get();
+            groupedHeaders.forEach(function(tdElement) {
+                $(tdElement).removeClass("sectorSorted");
+                $(tdElement).removeClass("sort_asc");
+                $(tdElement).removeClass("sort_desc");
+            });
+
+            var groupedSectors = $('.headersSector td').get();
+            groupedSectors.forEach(function(tdElement) {
+                $(tdElement).removeClass("sectorSorted");
+                $(tdElement).removeClass("sort_asc");
+                $(tdElement).removeClass("sort_desc");
+            });
+
+            if (sector == currentSortSector) {
+                reverseSector = !reverseSector;
+            } else {
+                reverseSector = false;
+                currentSortSector = sector;
+            }
+            var f = reverseSector ? 1 : -1;
+
+            $(groupedSectors[sector]).addClass("sectorSorted");
+            $(groupedSectors[sector]).addClass(reverseSector ? "sort_asc" : "sort_desc");
+
+            var rows = $('.technologiesPunchCard tbody tr').get();
+
+            rows.sort(function(a, b) {
+
+                var A = getVal(a);
+                var B = getVal(b);
+                if (!$.isNumeric(A) || !$.isNumeric(B)) {
+                    return B.localeCompare(A) * f;
+                }
+
+                if(A < B) {
+                    return -1*f;
+                }
+                if(A > B) {
+                    return 1*f;
+                }
+                return 0;
+            });
+
+            function getVal(elm) {
+                var v = $(elm).children('.sectorSummary').eq(sector).data("count");
+                if (v == null) {
+                    v = $(elm).children('.sectorSummary').eq(sector).text().trim();
+                } else if($.isNumeric(v)) {
+                    v = parseInt(v,10);
+                }
+                return v;
+            }
+
+            $.each(rows, function(index, row) {
+                $('.technologiesPunchCard').children('tbody').append(row);
+            });
+        }
+
         $().ready(function () {
             $(".headersGroup .sector").click(function (event) {
                 var td = event.target.parentNode;
```

Comment:
I noticed that every column of the header has a div inside, but because the div is shorter than the column(td) there is a problem when I click on the space between the div and the td (I'm attaching an image to explain this better)...

The problem is that whenever I click in the yellow part of the image the sort column is always the first one and not the column selected.

![screenshot from 2019-02-15 08-58-17](https://user-images.githubusercontent.com/2582866/52843023-1fd08b80-3101-11e9-8672-732c95ee4fa7.png)